core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java

```diff
@@ -586,15 +588,34 @@ private MethodDescriptor createRegisterSerializationForClassMethod(ClassCreator
         //register serialization feature as requested
         MethodCreator requiredFeatures = file.getMethodCreator("getRequiredFeatures", "java.util.List");
 
-        TryBlock requiredCatch = requiredFeatures.tryBlock();
+        TryBlock requiredTryCatch = requiredFeatures.tryBlock();
 
-        ResultHandle serializationFeatureClass = requiredCatch
-                .loadClassFromTCCL("com.oracle.svm.reflect.serialize.hosted.SerializationFeature");
-        ResultHandle requiredFeaturesList = requiredCatch.invokeStaticMethod(
+        AssignableResultHandle serializationFeatureClass = requiredTryCatch.createVariable(Class.class);
+
+        BranchResult graalVm22_3Test = requiredTryCatch.ifGreaterEqualZero(
+                requiredTryCatch.invokeVirtualMethod(VERSION_COMPARE_TO,
+                        requiredTryCatch.invokeStaticMethod(VERSION_CURRENT),
+                        requiredTryCatch.marshalAsArray(int.class, requiredTryCatch.load(22), requiredTryCatch.load(3))));
+        /* GraalVM >= 22.3 */
+        BytecodeCreator greaterThan22_2 = graalVm22_3Test.trueBranch();
+        greaterThan22_2.assign(serializationFeatureClass,
+                greaterThan22_2.loadClassFromTCCL("com.oracle.svm.hosted.reflect.serialize.SerializationFeature"));
+        /* GraalVM < 22.3 */
+        BytecodeCreator smallerThan22_3 = graalVm22_3Test.falseBranch();
+        smallerThan22_3.assign(serializationFeatureClass,
+                smallerThan22_3.loadClassFromTCCL("com.oracle.svm.reflect.serialize.hosted.SerializationFeature"));
+
+        ResultHandle requiredFeaturesList = requiredTryCatch.invokeStaticMethod(
                 ofMethod("java.util.Collections", "singletonList", List.class, Object.class),
                 serializationFeatureClass);
 
-        requiredCatch.returnValue(requiredFeaturesList);
+        requiredTryCatch.returnValue(requiredFeaturesList);
+
+        CatchBlockCreator cc = requiredTryCatch.addCatch(Throwable.class);
+        cc.invokeVirtualMethod(ofMethod(Throwable.class, "printStackTrace", void.class), cc.getCaughtException());
+        cc.throwException(cc.getCaughtException());
+
+        requiredFeatures.returnValue(requiredFeatures.loadNull());
```

Comment:
Do you actually need that? It looks extremely weird to me.

Reply:
Hmmm, you are right the `return null` is probably redundant.

Reply:
The print stack trace and rethrow is needed to make clear what the original exception was since GraalVM later catches it and prints a not that helpfull message. I guess I can/should fix GraalVM there as well.

Reply:
Exception handling fix is in https://github.com/oracle/graal/pull/4742

Shall i completely remove the catch block here?

Reply:
GitHub showed more context but my point was really about `requiredFeatures.returnValue(requiredFeatures.loadNull());`.